deep_quoridor/src/agents/alphazero/mcts.py

```diff
@@ -131,5 +131,7 @@ def search(self, initial_game: Quoridor):
                 self.new_nodes.extend(node.children)
 
             node.backpropagate(value)
-
-        return root.children
+        # Negate the value because the value is actually the value that the opponent
+        # got for getting to that state.
+        root_value = -(root.value_sum / root.visit_count)
```

Comment:
Shouldn't this only be negated if the opponent wins? I'm probably missing something.

Reply:
I spent a lot of time debugging this because initially I thought it was a bug. But actually, let's say that it's turn for player 1, so we have the root node, and every children will have a value saying how good this move is for player 1.   When you look at those children from the point of view of player 2 (because once you get into that children it will be player 2 turn), you need to negate it because when we computed it, it was from the pov of player 1.

Reply:
Ah, I see!